clause/label/label.go

```diff
@@ -0,0 +1,42 @@
+package label
+
+import (
+	"github.com/cyverse-de/querydsl"
+	"github.com/cyverse-de/querydsl/clause"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/olivere/elastic.v5"
+)
+
+const (
+	typeKey = "label"
+)
+
+var (
+	documentation = clause.ClauseDocumentation{
+		Summary: "Searches based on an object's label (typically, its filename)",
+		Args: map[string]clause.ClauseArgumentDocumentation{
+			"label": clause.ClauseArgumentDocumentation{Type: "string", Summary: "The label to search for"},
+		},
+	}
+)
+
+type LabelArgs struct {
+	Label string
+	// Negation bool // TODO
+	// Exact    bool // TODO
+}
+
+func LabelProcessor(args map[string]interface{}) (elastic.Query, error) {
+	var realArgs LabelArgs
+	err := mapstructure.Decode(args, &realArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	query := elastic.NewQueryStringQuery(realArgs.Label).Field("label")
+	return query, nil
+}
+
+func Register() {
```

Comment:
This is intended to be how consumers add this clause to the working set of clauses to be used. Once more clauses exist, there might also be helper packages for adding collections of clauses (e.g., a helper for everything that constitutes "data search") at once.